### PR TITLE
Restore on Linux.

### PR DESCRIPTION
**Pull request by niphlod**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [x] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Start with baby steps (support for Linux is experimental at this point, do we wanna issue a warning?!) for managing restore to linux instances. Yes, you still need Powershell to run on Windows. And yes, there will probably be some limitations. But... why not ?

### Approach
Start with realizing PS path management for anything outside of windows is ... lacking. 
Some things work out of the box, some don't. 
Some methods on Windows have a working counterpart on Linux's PS Core but... there's no way, e.g., to be on Windows and tell System.io.path to do get GetFullPath do the right thing "the unix way", or the other way around. 
Oh, it'll be fun times.
This first baby step introduces an internal function to realize path separator is not always '\'. Or, not anymore, at least.

### CODE REVIEW

**Comment by potatoqualitee** [comment]
just got one failing unit test

**Comment by niphlod** [comment]
Hu. I probably forgot to merge something else,.this is a reissue from past
week. Would probably be an easy fix but I'm eager to hear if the general
approach is sound

Il giorno mer 14 nov 2018, 01:09 Chrissy LeMaire <notifications@github.com>
ha scritto:

File: functions/Restore-DbaDatabase.ps1 [hunk 1]
```diff
@@ -290,16 +291,16 @@ function Restore-DbaDatabase {
         In this example we restore example1 database with no recovery, and then the second call is to set the database to recovery.
 
     .EXAMPLE
-        PS C:\> $SuspectPage = Get-DbaSuspectPage -SqlInstance server\instance1 -Database ProdFinance
-        PS C:\> Get-DbaBackupHistory - SqlInstance server\instance1 -Database -ProdFinance -Last | Restore-DbaDatabase -PageRestore PS C:\> $SuspectPage -PageRestoreTailFolder c:\temp -TrustDbBackupHistory -AllowContinues
+        PS C:\> Get-DbaBackupHistory - SqlInstance server\instance1 -Database -ProdFinance -Last | Restore-DbaDatabase -PageRestore
```
Comment (Stuart-Moore) [inline functions/Restore-DbaDatabase.ps1]:
```suggestion
        PS C:\> Get-DbaBackupHistory - SqlInstance server\instance1 -Database ProdFinance -Last | Restore-DbaDatabase -PageRestore
```

Comment (Stuart-Moore) [reply functions/Restore-DbaDatabase.ps1]: dbname appears to have morphed into a switch at some point in history

Comment (potatoqualitee) [reply functions/Restore-DbaDatabase.ps1]: thanks! fixed

**Pull request merged**